src/eckit/stats/comparator/ComparatorT.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <iosfwd>
+#include <limits>
+#include <string>
+
+#include "eckit/config/Parametrisation.h"
+#include "eckit/stats/Comparator.h"
+#include "eckit/stats/detail/CounterBinary.h"
+
+
+namespace eckit::stats {
+class Field;
+}
+
+
+namespace eckit::stats::comparator {
+
+
+/// Generic comparison on two Fields
+template <typename STATS>
+class ComparatorT : public Comparator, detail::CounterBinary, STATS {
```

Comment:
Because the implementation (and all concrete instantiations) are in the .cc file, I don't think this header file can be successfully included anywhere (well, certainly not if the contents are used).

Perhaps this template class definition would be better at the top of the .cc file...